### DISCUSSION TITLE
Binary data in responce

### DISCUSSION THREAD

**Discussion by IgorGrachov** [projectdiscovery/nuclei#4825] [Q&A]
I'm working with an HTTP GET request that retrieves a binary file. The response is being logged as a string starting with "HDR2" followed by a mix of escaped binary and Unicode characters, such as "HDR2\x00\x01\x0e\xdd\xd3\u0317\xb7\xa0\xa4\xa9\xae...".

I'm facing difficulties in preserving the integrity of the binary data when it's being saved to a log file. The issue arises when trying to convert or parse these logged representations back into their original binary form. No matter what method I try, I can't seem to restore the data to its correct state.

Is there a recommended way to log binary responses directly from GET requests within the Nuclei framework, or a preferred method to handle and convert these logged strings back into binary files?

Any guidance or suggestions would be greatly appreciated.

**Comment by Mzack9999** [comment]
@IgorGrachov could you provide the command (and template if possible) to reproduce this behavior? I'd be more than happy to help implementing the parser

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
